src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Service.cs

```diff
@@ -0,0 +1,377 @@
+            }
+
+            NNStreamer.CheckException(ret, "Failed to create ml_service instance");
+
+            _serviceEventCallback = (type, event_data_handle, data_handle) =>
+            {
+                if (type == EventType.NewData && _serviceEventReceived!= null) {
+                    MlInformation info = new MlInformation(event_data_handle);
+                    TensorsData data = TensorsData.CreateFromNativeHandle(data_handle, IntPtr.Zero, true, false);
+                    _serviceEventReceived?.Invoke(this, new ServiceReceivedEventArgs(info, data));
+                }
+            };
+        }
+
+        ~Service()
+        {
+            Dispose(false);
+        }
+
+        internal IntPtr GetHandle()
+        {
+            return _handle;
+        }
+
+
+        public void Start()
+        {
+            NNStreamerError ret = Interop.Service.Start(_handle);
+            NNStreamer.CheckException(ret, "Failed to start the service because of internal error");
+        }
+
+        public void Stop()
+        {
+            NNStreamerError ret = Interop.Service.Stop(_handle);
+            NNStreamer.CheckException(ret, "Failed to stop the service because of internal error");
+        }
+
+        public TensorsInfo GetTensorsInformation(string name, bool is_input)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            IntPtr inHandle;
+            NNStreamerError ret;
+
+            if (is_input)
+                ret = Interop.Service.GetInputInformation(_handle, name, out inHandle);
+            else
+                ret = Interop.Service.GetOutputInformation(_handle, name, out inHandle);
+
+            NNStreamer.CheckException(ret, "Failed to get TensorsInfo handle");
+
+            TensorsInfo retInfo = TensorsInfo.ConvertTensorsInfoFromHandle(inHandle);
+            return retInfo;
+        }
+
+        public void SetInformation(string name, string value)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            if (string.IsNullOrEmpty(value))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property value is invalid");
+
+            NNStreamerError ret = Interop.Service.SetInformation(_handle, name, value);
+            NNStreamer.CheckException(ret, "Failed to set service information");
+        }
+
+        public string Get(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            string value;
+            NNStreamerError ret = Interop.Service.GetInformation(_handle, name, out value);
```

Comment:
need marshaling?
1. change param interop GetInformation (matched with ml_service_get_information)
    out  IntPtr value
2. return string
```
IntPtr value = IntPtr.Zero;
NNStreamerError ret = Interop.Service.GetInformation(_handle, name, out value);
NNStreamer.CheckException(ret, "Failed to get service information");

return Interop.Util.IntPtrToString(value);
```

Reply:
Thank you :)

###  reference
```cs
        /* int ml_single_get_property (ml_single_h single, const char *name, char **value) */
        [DllImport(Libraries.MlSingle, EntryPoint = "ml_single_get_property", CallingConvention = CallingConvention.Cdecl)]
        internal static extern NNStreamerError GetValue(IntPtr single_handle, string name, out IntPtr value);
```

Reply:
Updated!